### PR TITLE
python3Packages.pygame: 2.0.1 -> 2.1.0

### PR DESCRIPTION
**Pull request by emilytrau**
###### Motivation for this change

Bumps pygame from 2.0.1 to the first 2.0.2 pre-release. This is the first release in over a year and contains significant improvements. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 136009` at b95c5d0f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>impressive</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/d3dd68e4d074b8883f74080bb6c6c6e4">python38Packages.rl-coach</a></li>
    <li><a href="https://gist.github.com/d911575d4c405685f08771c77f18116c">python39Packages.rl-coach</a></li>
  </ul>
</details>
<details>
  <summary>16 packages built successfully:</summary>
  <ul>
    <li>_20kly</li>
    <li>brutalmaze</li>
    <li>cantoolz</li>
    <li>endgame-singularity</li>
    <li>frescobaldi</li>
    <li>gnome.aisleriot</li>
    <li>pysolfc</li>
    <li>python38Packages.mido</li>
    <li>python38Packages.pygame</li>
    <li>python38Packages.pygame-gui</li>
    <li>python38Packages.pytmx</li>
    <li>python39Packages.mido</li>
    <li>python39Packages.pygame</li>
    <li>python39Packages.pygame-gui</li>
    <li>python39Packages.pytmx</li>
    <li>video2midi</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/pygame/default.nix:66:0:
```
   |
66 |     description = "Python library for games";
   | ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by Artturin** [comment]
there's a newer release available, could you update the pr?

**Comment by angustrau** [comment]
I've updated this to the latest release. An SDL2_mixer update added timidity as a dependency so I added darwin support for that.

**Comment by angustrau** [comment]
Result of `nixpkgs-review pr 136009` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>impressive</li>
    <li>libsForQt512.soundkonverter</li>
    <li>libsForQt514.soundkonverter</li>
    <li>python38Packages.rl-coach</li>
    <li>python39Packages.rl-coach</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>appimage-run-tests</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>frogatto</li>
    <li>worldofgoo</li>
  </ul>
</details>
<details>
  <summary>133 packages built:</summary>
  <ul>
    <li>MMA</li>
    <li>SDL2_mixer</li>
    <li>Sylk</li>
    <li>_1oom</li>
    <li>_20kly</li>
    <li>abbaye-des-morts</li>
    <li>altair</li>
    <li>anki-bin</li>
    <li>anytype</li>
    <li>appimage-run</li>
    <li>apple-music-electron</li>
    <li>augustus</li>
    <li>beekeeper-studio</li>
    <li>betterdiscord-installer</li>
    <li>blobwars</li>
    <li>brutalmaze</li>
    <li>buttercup-desktop</li>
    <li>cantoolz</li>
    <li>cataclysm-dda</li>
    <li>cataclysm-dda-git</li>
    <li>cdogs-sdl</li>
    <li>chocolateDoom</li>
    <li>chromium-bsu</li>
    <li>chrysalis</li>
    <li>commandergenius</li>
    <li>cozy-drive</li>
    <li>craftos-pc</li>
    <li>crawlTiles</li>
    <li>crispyDoom</li>
    <li>crypto-org-wallet</li>
    <li>devdocs-desktop</li>
    <li>devilutionx</li>
    <li>easyrpg-player</li>
    <li>eduke32</li>
    <li>electron-mail</li>
    <li>electronplayer</li>
    <li>endgame-singularity</li>
    <li>enigma</li>
    <li>fheroes2</li>
    <li>flare</li>
    <li>flexoptix-app</li>
    <li>frescobaldi</li>
    <li>gemrb</li>
    <li>gnome.aisleriot</li>
    <li>guile-sdl2</li>
    <li>haskellPackages.Raincat</li>
    <li>haskellPackages.piyo</li>
    <li>haskellPackages.sdl2-mixer</li>
    <li>hedgewars</li>
    <li>instead</li>
    <li>instead-launcher</li>
    <li>irccloud</li>
    <li>ivan</li>
    <li>joplin-desktop</li>
    <li>julius</li>
    <li>jumpnbump</li>
    <li>keeweb</li>
    <li>lbry</li>
    <li>ledger-live-desktop</li>
    <li>lens</li>
    <li>liberal-crime-squad</li>
    <li>soundkonverter (libsForQt5.soundkonverter)</li>
    <li>lightspark</li>
    <li>linuxband</li>
    <li>losslesscut-bin</li>
    <li>lunar-client</li>
    <li>manaplus</li>
    <li>mar1d</li>
    <li>marktext</li>
    <li>mathpix-snipping-tool</li>
    <li>methane</li>
    <li>midisheetmusic</li>
    <li>moc</li>
    <li>molotov</li>
    <li>mycrypto</li>
    <li>neo4j-desktop</li>
    <li>notable</li>
    <li>np2kai</li>
    <li>nuclear</li>
    <li>nxengine-evo</li>
    <li>octant-desktop</li>
    <li>opendune</li>
    <li>p3x-onenote</li>
    <li>pingus</li>
    <li>playonlinux</li>
    <li>plexamp</li>
    <li>pysolfc</li>
    <li>python38Packages.kivy</li>
    <li>python38Packages.mido</li>
    <li>python38Packages.pygame</li>
    <li>python38Packages.pygame-gui</li>
    <li>python38Packages.pygame_sdl2</li>
    <li>python38Packages.pysdl2</li>
    <li>python38Packages.pytmx</li>
    <li>python39Packages.kivy</li>
    <li>python39Packages.mido</li>
    <li>python39Packages.pygame</li>
    <li>python39Packages.pygame-gui</li>
    <li>python39Packages.pygame_sdl2</li>
    <li>python39Packages.pysdl2</li>
    <li>python39Packages.pytmx</li>
    <li>radicle-upstream</li>
    <li>rambox-pro</li>
    <li>renpy</li>
    <li>retrofe</li>
    <li>rocksndiamonds</li>
    <li>ryujinx</li>
    <li>saleae-logic-2</li>
    <li>session-desktop-appimage</li>
    <li>sfrotz</li>
    <li>solfege</li>
    <li>ssb-patchwork</li>
    <li>standardnotes</li>
    <li>station</li>
    <li>steam-run-native</li>
    <li>steam-tui</li>
    <li>taisei</li>
    <li>tauon</li>
    <li>timeular</li>
    <li>timidity</li>
    <li>trackballs</li>
    <li>trezor-suite</li>
    <li>tusk</li>
    <li>ultrastardx</li>
    <li>unityhub</li>
    <li>via</li>
    <li>vial</li>
    <li>video2midi</li>
    <li>wesnoth (wesnoth-dev)</li>
    <li>wootility</li>
    <li>ytmdesktop</li>
    <li>zettlr</li>
    <li>zulip</li>
  </ul>
</details>

**Comment by angustrau** [comment]
As I put this up they've released a 2.1.0 dev preview. This next release will remove python 2 support. Probably best to wait till after 21.11 for that?

**Comment by SuperSandro2000** [comment]
> As I put this up they've released 2.1.0. This next release is primarily to remove python 2 support. Probably best to wait till after 21.11 for that?

Doesn't matter. python2 is mostly broken except for nixops anyway and already support only the bare minimum for 21.11.

File: pkgs/tools/misc/timidity/default.nix [hunk 1]
```diff
@@ -29,12 +44,14 @@ stdenv.mkDerivation rec {
     cp ${./timidity.cfg} $out/share/timidity/timidity.cfg
     tar --strip-components=1 -xf $instruments -C $out/share/timidity/
   '';
+  # This fixup step is unnecessary and fails on Darwin
+  dontRewriteSymlinks = true;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/timidity/default.nix:48]:
```suggestion
  dontRewriteSymlinks = stdenv.isDarwin;
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 136009` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages marked as broken and skipped:
</summary>
<ul>
<li>impressive</li>
<li>libsForQt512.soundkonverter</li>
<li>libsForQt514.soundkonverter</li>
<li>python38Packages.rl-coach</li>
<li>python39Packages.rl-coach</li>
</ul>
</details>
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>appimage-run-tests</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>worldofgoo: log was empty</li>
  </ul>
</details>
<details>

  <summary>15 packages failed to build and are new build failures:</summary>
  <ul>
<li>anytype: log was empty</li>
<li>endgame-singularity: log was empty</li>
<li>flexoptix-app: log was empty</li>
<li>frogatto: log was empty</li>
<li>gnome.aisleriot: log was empty</li>
<li>hedgewars: log was empty</li>
<li>notable: log was empty</li>
<li>nxengine-evo: log was empty</li>
<li>plexamp: log was empty</li>
<li>pysolfc: log was empty</li>
<li>rambox-pro: log was empty</li>
<li>rocksndiamonds: log was empty</li>
<li>saleae-logic-2: log was empty</li>
<li>trezor-suite: log was empty</li>
<li>via: log was empty</li>
  </ul>
</details>
<details>
<summary>
118 packages built:
</summary>
<ul>
<li>MMA</li>
<li>SDL2_mixer</li>
<li>Sylk</li>
<li>_1oom</li>
<li>_20kly</li>
<li>abbaye-des-morts</li>
<li>altair</li>
<li>anki-bin</li>
<li>appimage-run</li>
<li>apple-music-electron</li>
<li>augustus</li>
<li>beekeeper-studio</li>
<li>betterdiscord-installer</li>
<li>blobwars</li>
<li>brutalmaze</li>
<li>buttercup-desktop</li>
<li>cantoolz</li>
<li>cataclysm-dda</li>
<li>cataclysm-dda-git</li>
<li>cdogs-sdl</li>
<li>chocolateDoom</li>
<li>chromium-bsu</li>
<li>chrysalis</li>
<li>commandergenius</li>
<li>cozy-drive</li>
<li>craftos-pc</li>
<li>crawlTiles</li>
<li>crispyDoom</li>
<li>crypto-org-wallet</li>
<li>devdocs-desktop</li>
<li>easyrpg-player</li>
<li>eduke32</li>
<li>electron-mail</li>
<li>electronplayer</li>
<li>enigma</li>
<li>fheroes2</li>
<li>flare</li>
<li>frescobaldi</li>
<li>gemrb</li>
<li>guile-sdl2</li>
<li>haskellPackages.Raincat</li>
<li>haskellPackages.piyo</li>
<li>haskellPackages.sdl2-mixer</li>
<li>instead</li>
<li>instead-launcher</li>
<li>irccloud</li>
<li>ivan</li>
<li>joplin-desktop</li>
<li>julius</li>
<li>jumpnbump</li>
<li>keeweb</li>
<li>lbry</li>
<li>ledger-live-desktop</li>
<li>lens</li>
<li>liberal-crime-squad</li>
<li>soundkonverter (libsForQt5.soundkonverter)</li>
<li>lightspark</li>
<li>linuxband</li>
<li>losslesscut-bin</li>
<li>lunar-client</li>
<li>manaplus</li>
<li>mar1d</li>
<li>marktext</li>
<li>mathpix-snipping-tool</li>
<li>methane</li>
<li>midisheetmusic</li>
<li>moc</li>
<li>molotov</li>
<li>mycrypto</li>
<li>neo4j-desktop</li>
<li>np2kai</li>
<li>nuclear</li>
<li>octant-desktop</li>
<li>opendune</li>
<li>p3x-onenote</li>
<li>pingus</li>
<li>playonlinux</li>
<li>python38Packages.kivy</li>
<li>python38Packages.mido</li>
<li>python38Packages.pygame</li>
<li>python38Packages.pygame-gui</li>
<li>python38Packages.pygame_sdl2</li>
<li>python38Packages.pysdl2</li>
<li>python38Packages.pytmx</li>
<li>python39Packages.kivy</li>
<li>python39Packages.mido</li>
<li>python39Packages.pygame</li>
<li>python39Packages.pygame-gui</li>
<li>python39Packages.pygame_sdl2</li>
<li>python39Packages.pysdl2</li>
<li>python39Packages.pytmx</li>
<li>radicle-upstream</li>
<li>renpy</li>
<li>retrofe</li>
<li>ryujinx</li>
<li>session-desktop-appimage</li>
<li>sfrotz</li>
<li>solfege</li>
<li>ssb-patchwork</li>
<li>standardnotes</li>
<li>station</li>
<li>steam-run-native</li>
<li>steam-tui</li>
<li>taisei</li>
<li>tauon</li>
<li>timeular</li>
<li>timidity</li>
<li>trackballs</li>
<li>tusk</li>
<li>ultrastardx</li>
<li>unityhub</li>
<li>vial</li>
<li>video2midi</li>
<li>wesnoth (wesnoth-dev)</li>
<li>wootility</li>
<li>ytmdesktop</li>
<li>zettlr</li>
<li>zulip</li>
</ul>
</details>

**Pull request merged**